lib/spack/spack/solver/asp.py

```diff
@@ -3065,6 +3065,10 @@ def setup(
         self.possible_compilers = list(candidate_compilers)
         self.possible_compilers.sort()  # type: ignore[call-overload]
 
+        should_mix = 1 if spack.config.get("concretizer:compiler_mixing", True) else 0
+        for lang in ["c", "cxx", "fortran"]:
+            self.gen.fact(fn.compiler_mixing(should_mix, lang))
```

Comment:
I would make this fact unary, call it something like:
```python
no_compiler_mixing(lang)
```
and emit it only if compiler mixing is disabled. Seeing a fact with a hard-coded `0` in the ASP is not very legible imo.